pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+            if numpy.any(numpy.array(order)[srt]-1 != numpy.arange(order[srt[-1]])):
+                raise ValueError('Parameter set series must be sequential and 1-indexed.')
+            return [par[i] for i in srt]
+        return None
+
+#-----------------------------------------------------------------------------
+# Reduction ParSets
+
+class FrameGroupPar(ParSet):
+    def __init__(self, frametype=None, useframe=None, number=None, combine=None):
+        # Grab the parameter names and values from the function
+        # arguments
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        pars = dict([(k,values[k]) for k in args[1:]])
+
+        # Initialize the other used specifications for this parameter
+        # set
+        defaults = dict.fromkeys(pars.keys())
+        options = dict.fromkeys(pars.keys())
+        dtypes = dict.fromkeys(pars.keys())
+        descr = dict.fromkeys(pars.keys())
+
+        # Fill out parameter specifications.  Only the values that are
+        # *not* None (i.e., the ones that are defined) need to be set
+        defaults['frametype'] = 'bias'
+        options['frametype'] = FrameGroupPar.valid_frame_types()
+        dtypes['frametype'] = str
+        descr['frametype'] = 'Frame type.  ' \
+                             'Options are: {0}'.format(', '.join(options['frametype']))
+
+        dtypes['useframe'] = str
+        descr['useframe'] = 'A master calibrations file to use if it exists'
+
+        defaults['number'] = 0
+        dtypes['number'] = int
+        descr['number'] = 'Number of frames to use of this type'
+
+        defaults['combine'] = CombineFramesPar()
+        dtypes['combine'] = [ParSet, dict]
+        descr['combine'] = 'Parameters used when combining frames of this type'
+
+        # Instantiate the parameter set
+        super(FrameGroupPar, self).__init__(list(pars.keys()),
+                                            values=list(pars.values()),
+                                            defaults=list(defaults.values()),
+                                            options=list(options.values()),
+                                            dtypes=list(dtypes.values()),
+                                            descr=list(descr.values()))
+
+        self.validate()
+
+    @classmethod
+    def from_dict(cls, frametype, cfg):
+        k = cfg.keys()
+        parkeys = [ 'useframe', 'number' ]
+        kwargs = {}
+        for pk in parkeys:
+            kwargs[pk] = cfg[pk] if pk in k else None
+        pk = 'combine'
+        kwargs[pk] = CombineFramesPar.from_dict(cfg[pk]) if pk in k else None
+        return cls(frametype=frametype, **kwargs)
+
+    @staticmethod
+    def valid_frame_types():
+        """
+        Return the list of valid frame types.
+        """
+        return [ 'bias', 'pixelflat', 'arc', 'pinhole', 'trace', 'standard', 'science' ]
+
+    def validate(self):
+        pass
+
+
+class CombineFramesPar(ParSet):
```

Comment:
I've also managed to munge this capability in PYPIT.
But we could bring it back.

Reply:
I should say the matching.
Not the combining.